frontend/src/components/SolvedTable/SolvedTable.component.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from "react";
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  TableCaption,
+  Center,
+  TableContainer,
+} from "@chakra-ui/react";
+import { Paginator } from "../Paginator/Paginator.component";
+import { SolvedQuestion } from "../../models/SolvedQuestion.model";
+import { fetchUserCompletedQuestions } from "../../api/user"; // Import your API function
+
+export type TableProp = {
+  userId: number;
+  pageSize?: number;
+};
+
+export const SolvedTable = (props: TableProp) => {
+  const { userId, pageSize = 10 } = props;
+  const [currentPage, setCurrentPage] = useState(1);
+  const [solvedQuestions, setSolvedQuestions] = useState<SolvedQuestion[]>([]);
+
+  const onPageChange = (page: React.SetStateAction<number>) => {
+    setCurrentPage(page);
+  };
+
+  useEffect(() => {
+  const loadSolvedQuestions = async () => {
+    if (userId) {
+      try {
+        const data = await fetchUserCompletedQuestions(userId);
+        setSolvedQuestions(data);
+      } catch (error) {
+        console.error("Failed to fetch solved questions:", error);
+      }
+    }
+  };
```

Comment:
any reason why you define this method inside the useEffect hook?

I think it can be defined outside the useEffect hook

Reply:
By defining the loadSolvedQuestions function inside the useEffect, I thought I would limit the scope of the function to this specific effect.